packages/terra-clinical-action-header/src/ActionHeader.jsx

```diff
@@ -0,0 +1,171 @@
+   * On small viewports a back button will be displayed instead of a close button when a separate onBack callback is not set.
+   **/
+  onClose: PropTypes.func,
+
+  /**
+   * Callback function for when the back button is clicked. The back button will not display if this is not set.
+   **/
+  onBack: PropTypes.func,
+
+  /**
+   * Callback function for when the expand button is clicked.
+   * The expand button will not display if this is not set or on small viewports.
+   * Only the expand button will be rendered if onExpand and onMinimize are set.
+   **/
+  onExpand: PropTypes.func,
+
+  /**
+   * Callback function for when the minimize button is clicked.
+   * The minimize button will not display if this is not set or on small viewports.
+   * Only the expand button will be rendered if both onExpand and onMinimize are set.
+   **/
+  onMinimize: PropTypes.func,
+
+  /**
+   * Callback function for when the next button is clicked. The previous-next button group will not display if neither this or onPrevious are set.
+   **/
+  onNext: PropTypes.func,
+
+  /**
+   * Callback function for when the previous button is clicked. The previous-next button group will not display if neither this or onNext are set.
+   **/
+  onPrevious: PropTypes.func,
+
+  /**
+   * Child element to be displayed on the right end of the header.
+   **/
+  children: PropTypes.element,
+};
+
+const defaultProps = {
+  title: '',
+  onClose: null,
+  onBack: null,
+  onExpand: null,
+  onMinimize: null,
+  onNext: null,
+  onPrevious: null,
+  children: null,
+};
+
+const ActionHeader = ({
+  title,
+  onClose,
+  onBack,
+  onExpand,
+  onMinimize,
+  onPrevious,
+  onNext,
+  children,
+  ...customProps
+}) => {
+  const attributes = Object.assign({}, customProps);
+  let closeButton = onClose ? <Button icon={<IconClose />} onClick={onClose} /> : null;
+  let backButton = onBack ? <Button icon={<IconLeft />} onClick={onBack} /> : null;
+
+  let expandButton;
+  if (!backButton) {
+    if (onExpand) {
+      expandButton = <Button icon={<IconMaximize />} onClick={onExpand} />;
+    } else if (onMinimize) {
+      expandButton = <Button icon={<IconMinimize />} onClick={onMinimize} />;
+    }
+  }
+
+  let previousNextButtonGroup = null;
+  if (onPrevious || onNext) {
+    previousNextButtonGroup = (
+      <ButtonGroup>
+        <ButtonGroup.Button icon={<IconChevronUp />} onClick={onPrevious} />
+        <ButtonGroup.Button icon={<IconChevronDown />} onClick={onNext} />
+      </ButtonGroup>
+    );
+  }
+
+  const leftButtonsDefault = (
+    <div className="terraClinical-ActionHeader-leftButtons">
+      {backButton}
+      {expandButton}
+      {previousNextButtonGroup}
+    </div>
+  );
+
+  const rightButtonsDefault = (
+    <div className="terraClinical-ActionHeader-rightButtons">
+      {children}
+      {closeButton}
+    </div>
+  );
+
+  if (onClose && !onBack) {
```

Comment:
This was here to modify the close in back buttons for the small viewport case. I went ahead and refactored this a bit so it would be easier understand what was going on: https://github.com/cerner/terra-clinical/pull/28/commits/bbedd6990687be8e80790cf27ef524f565b492ae